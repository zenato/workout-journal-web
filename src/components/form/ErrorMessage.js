import React from 'react';
import { getErrorMessages } from 'lib/api';

const ErrorMessage = ({ error, name, context = 'data' }) => {
  const errorMessages = getErrorMessages(error, name, context);
  return errorMessages ? (
    <ul className="error-message">
      {errorMessages.map((message, idx) => <li key={idx}>{message}</li>)}
    </ul>
  ) : null;
};

export default ErrorMessage;
