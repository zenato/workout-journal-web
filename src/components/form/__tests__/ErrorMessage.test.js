import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from '../ErrorMessage';

it('renders error', () => {
  const fieldName = 'name';
  const error = {
    data: {
      [fieldName]: ['error-1l', 'error-2'],
    },
  };
  const wrapper = shallow(<ErrorMessage name={fieldName} error={error} />);
  expect(wrapper).toMatchSnapshot();
});