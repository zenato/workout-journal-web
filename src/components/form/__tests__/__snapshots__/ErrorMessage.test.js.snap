// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="error-message"
>
    <li>
        error-1l
    </li>
    <li>
        error-2
    </li>
</ul>,
  "nodes": Array [
    <ul
      className="error-message"
>
      <li>
            error-1l
      </li>
      <li>
            error-2
      </li>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ErrorMessage
        error={
                Object {
                        "data": Object {
                          "name": Array [
                            "error-1l",
                            "error-2",
                          ],
                        },
                      }
        }
        name="name"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": Object {
            "data": Object {
              "name": Array [
                "error-1l",
                "error-2",
              ],
            },
          },
          "name": "name",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="error-message"
>
          <li>
                    error-1l
          </li>
          <li>
                    error-2
          </li>
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="error-message"
>
          <li>
                    error-1l
          </li>
          <li>
                    error-2
          </li>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ErrorMessage
    error={
        Object {
            "data": Object {
              "name": Array [
                "error-1l",
                "error-2",
              ],
            },
          }
    }
    name="name"
/>,
}
`;
